### DISCUSSION TITLE
InnerTune now has trackers

### DISCUSSION THREAD

**Discussion by M00NJ** [z-huang/music#941] [Polls]
The latest release has introduced proprietary tracking libraries to InnerTune. I'm disappointed that this was not mentioned in the release notes, but thankfully it was brought to my attention by IzzySoft in #935. The developer of IT has already stated their reasons behind this decision:

"Many people don't know how to get adb logs, so I can't fix some issues. Firebase Crashlytics records logs automatically, which helps me fix bugs more efficiently. Also, I plan to add some new features, and I need to know the geographic distribution of my users. Therefore, I added Firebase Analytics. Firebase provides many other features which can be utilized in the future.
The goal of this app is to provide a free music streaming service as good as YouTube Premium, even better. To be honest, compared with better user experience, privacy isn't my first concern. Even though there's no trackers in the previous versions, requests to music.youtube.com make it possible for YouTube to track you. I know that F-Droid and IzzyOnDroid don't allow apps with trackers, so I'd like to request a removal of this app in these platform. Users can only download this app in GitHub. But no worries, the app will automatically check for updates and notify users.
Anyways, @IzzySoft, thank you for discovering and promoting this app when it's in the early stage."

I want to start a discussion about this. Here are some questions to start off this topic:

- Are you okay with trackers?
- Do you want your music app to use your geolocation?
- Do you mind if InnerTune gets removed from IzzyOnDroid and F-Droid?
- Will you keep using InnerTune with trackers in it?

I myself have already commented in the original issue, so I will just paste this here:

"I have no say here just wanted to give feedback. I'm very unhappy with this development direction. I use InnerTuner FOR privacy reasons. Although I think having the goal to improve the app's user experience to a point where it's as good or better than YT Music is great, I disagree that this has to come at the expense of the user's privacy. I don't see how knowing the users geographical distribution would be necessary to further this goal and I doubt that the data provided by this change would be of much use. The open source community is very passionate about privacy and many use VPNs or something like orbot to hide their location. The argument, that the requests to music.youtube.com already make the user vulnerable to tracking is only true if the user doesn't take steps to prevent this (for example by using a VPN again). I also think removing the app from Izzy and F-droid will greatly decrease it's discoverability and therefore bring down the amount of contributions made to this project. So to me the only argument that holds real weight is the efficiency gain in fixing bugs, because of the automatically recorded log's. But I'm sure there are alternatives to this and also think the loss of users due to the removal of the app from popular open source repositories and because many current users will stop using and contributing to the app, because they are unhappy with these changes, will cancel out this advantage over time. I'm sure you've thought this through very thoroughly and won't change you mind because of some random comment, but I hope I can convince you to at least wait for the community to have a discussion about this, before going through with the removal of IT from Izzy and F-droid."

Feel free to vote in the poll below, so we can have a visual representation of how people feel about this.

**Comment by IzzySoft** [comment]
As I'm not using the app myself, I won't participate in the poll – but you could count my vote into the last option: I wouldn't even give it a try (i.e. I would not install the app in the first place) as long as those libraries are present. There are many good alternatives available coming without those proprietary parts, so I'd go for them first. And would probably stick with one of those as they fit my needs (having NewPipe on a tablet, for example, and see no need to switch to something else).

That said, and as outlined by you already, there are alternatives. And I'd much favor we find a solution here (e.g. the FOSS build) over having to remove the app from F-Droid and/or my repo. People should have choices, also when deciding on what app to use – so it's good to have a variety of apps available on a given topic :wink:

**Comment by bearss69** [comment]
I use innertune every day. Other apps that I have tried are simply not as reliable or have as many features as innertune, i.e., spotube, musify, etc. And I'm not too concerned about trackers either, so I'll just hope it gets fixed as soon as possible and just keep using it just like how I've been using it in the meantime.

**Reply by DeadSecTr** [reply]
Yes....but without trackers! Thats why we use/want opensource apps
